core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -344,13 +357,21 @@ public void write(byte[] b, int off, int len) throws IOException {
         writeBytes(b, off, len);
     }
 
-    public void writeStringArray(String[] array) throws IOException {
+    public void writeStringArray(String... array) throws IOException {
         writeVInt(array.length);
         for (String s : array) {
             writeString(s);
         }
     }
 
+    public void writeStringArray(List<String> strings) throws IOException {
+        writeVInt(strings.size());
+        for (String s : strings) {
+            writeString(s);
+        }
+    }
```

Comment:
I dont know about this one...feels like if we add this we should add other kagillion similar methods (e.g. List<Integer>, List<Whatever>), which is not best.